gammapy/irf/tests/test_background.py

```diff
@@ -40,4 +41,10 @@ def test_background_3d_basics(bkg_3d):
 def test_background_3d_evalutate(bkg_3d):
     bkg_rate = bkg_3d.data.evaluate(energy='1 TeV', detx='0.2 deg', dety='0.5 deg')
     assert_allclose(bkg_rate.value, 0.00013652553025167435)
-    bkg_rate.unit == u.Unit('s-1 MeV-1 sr-1')
+    assert bkg_rate.unit == u.Unit('s-1 MeV-1 sr-1')
+
+@requires_data('gammapy-extra')
+def test_background3d_write(bkg_3d):
+    hdu =  table_to_fits_table(bkg_3d.to_table())
+    assert hdu.data['DETX_LO'][0].all() == bkg_3d.data.axis('detx').lo.value.all()
```

Comment:
@thomasarmstrong - Why do you call `.all()` here? I think what that does is to convert the data arrays to bool and check if all entries are 0 or non-0. This is not a useful assert, no? Did you mean this?
```
assert hdu.data['DETX_LO'][0] == bkg_3d.data.axis('detx').lo.value
```
I think that would be a useful assert and work as long as the array dtypes are the same, i.e. no float64 to float32 conversions occur. I.e. bit-by-bit comparison should work here, but if it doesn't, then you should call `numpy.testing.assert_allclose`.